### PR TITLE
Add 'return-error-limit' option to limit the number of returned matched lines\log entries

### PR DESCRIPTION
**Pull request by jhshadi**
## Pull Request Checklist

**Is this in reference to an existing issue?**
#28 

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose
limit the number of returned matched lines\log entries.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,8 +25,8 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
 
 ## [1.1.0] - 2017-05-20
 ### Added
-- Add `return-length` option to support custom length for returned matched lines
-- Add `log-pattern` option to support read and match against whole log entry instead of single line
+- Add `return-length` option to support custom length for returned matched lines (@jhshadi)
```
Comment (majormoses) [inline CHANGELOG.md:28]: are you shlomi? https://github.com/sensu-plugins/sensu-plugins-logs/compare/1.0.0...1.1.0 could you please confirm something that we can verify to give you credit?

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,8 +25,8 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
...
+- Add `log-pattern` option to support read and match against whole log entry instead of single line (@jhshadi)
```
Comment (majormoses) [inline CHANGELOG.md:29]: are you shlomi? https://github.com/sensu-plugins/sensu-plugins-logs/compare/1.0.0...1.1.0 could you please confirm something that we can verify to give you credit?

File: bin/check-log.rb [hunk 3]
```diff
@@ -128,6 +128,11 @@ class CheckLog < Sensu::Plugin::Check::CLI
          proc: proc(&:to_i),
          default: 250
 
+  option :return_error_limit,
+         description: 'Max number of returned matched lines\log entries',
```
Comment (majormoses) [inline bin/check-log.rb]: if you want to call out log entries then put it on parenthesis they are not equivalent. Not all logs always are single lines.

**Review by majormoses — CHANGES_REQUESTED** [review]
Overall looks good, I need some verification of a few things:
- git ownership of commits asking for credit
- testing artifact: this can be a unit test or a gist/screenshot of the input and output to make sure it works as intended.

File: CHANGELOG.md (hunk 2 above)
Comment (jhshadi) [reply CHANGELOG.md:29]: @majormoses sure, what\how to confirm it?

Comment (jhshadi) [reply CHANGELOG.md:29]: @majormoses I have tested it locally but i can add a unit test.
Can you reference to an example test and will take it from there?

Comment (majormoses) [reply CHANGELOG.md:29]: do you still have access to this email? shlomi@innovid.com

Comment (jhshadi) [reply CHANGELOG.md:29]: @majormoses yes, i do have access to shlomi@innovid.com.

**Comment by majormoses** [comment]
regarding testing just running the command locally and capturing the input and output in a gist is enough, something like this: https://github.com/sensu-plugins/sensu-plugins-elasticsearch/pull/16 regarding unit tests there are some decent examples in the mesos or aws plugins.

File: CHANGELOG.md (hunk 2 above)
Comment (majormoses) [reply CHANGELOG.md:29]: I will send you an email and have you confirm that is enough for me.

You will need to reply back with the following: "I really need to setup github with the email address outlined here: https://help.github.com/articles/setting-your-commit-email-address-on-github/"

Comment (majormoses) [reply CHANGELOG.md:29]: confirmed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
 
 ## [Unreleased]
 ### Added
-- Add `return-error-limit` option to limit the number of returned matched lines\log entries (@jhshadi)
+- Add `return-error-limit` option to limit the number of returned matched lines\"log entries" (@jhshadi)
```
Comment (majormoses) [inline CHANGELOG.md]: either actually use parenthesis or change the commit message to match whats being changed here...

**Review by majormoses — CHANGES_REQUESTED** [review]

**Comment by majormoses** [comment]
@jhshadi any plans to pick this back up?

**Comment by majormoses** [comment]
closing due to inactivity

**Comment by jhshadi** [comment]
@majormoses sorry for the long absent, can you reopen the pull request?

**Comment by jhshadi** [comment]
@majormoses accidentally reverted the change and then apply it back. sorry for the hustle :)
in any case i run the command locally with the `--return-error-limit` feature:
```bash
$ ./bin/check-log.rb --log-file test/logs/example_01.log --state-dir ./test/logs/state --pattern error --crit 1 --return-error-limit 1 --return
CheckLog CRITICAL: 0 warnings, 2 criticals for pattern error. 
this is error line
```
tried adding unit tests using the reference you linked above but couldn't make them run properly. will try adding those later on.

LMK if there's anything else is needed.

**Comment by jhshadi** [comment]
@majormoses any other changes needed here?

**Comment by majormoses** [comment]
sorry thought I got back to this one. Let me review and will report back shortly.

**Review by majormoses — APPROVED** [review]

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-logs/versions/1.3.0

**Pull request merged**